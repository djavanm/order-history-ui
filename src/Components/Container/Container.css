section {
  min-height: 700px;
  background:#CECECE;
  display: flex;
  flex-direction: column;
}