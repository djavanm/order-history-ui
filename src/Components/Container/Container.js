import React from 'react';
import './Container.css';
import Card from '../Card/Card.js'

const Container = ({ purchases, deleteOldPurchase }) => {
  const purchaseCards = purchases.map(purchase => {
    return <Card purchase={purchase} key={purchase.id} deleteOldPurchase={deleteOldPurchase} />
  })  
  return (
    <section>
      {purchaseCards}
    </section>  
  )
}

export default Container;