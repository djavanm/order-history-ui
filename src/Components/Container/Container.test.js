import React from 'react';
import Container from './Container';
import { shallow } from 'enzyme';

describe('Container' , () => {
  let mockData;
  beforeEach(() => {
    mockData = {
      id: 1,
      name: 'Djavan',
      price: 9999, 
      description: 'Great for the whole family',
      img: 'random corner of the internet'
    }
  })
  it('should match the snapshot with the mock data', () => {
    const wrapper = shallow(<Container purchases={[mockData]}
    />)
    expect(wrapper).toMatchSnapshot();
  })
});