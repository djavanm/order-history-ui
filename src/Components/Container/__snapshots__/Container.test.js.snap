// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match the snapshot with the mock data 1`] = `
<section>
  <Card
    key="1"
    purchase={
      Object {
        "description": "Great for the whole family",
        "id": 1,
        "img": "random corner of the internet",
        "name": "Djavan",
        "price": 9999,
      }
    }
  />
</section>
`;
